app/src/main/java/org/mozilla/fenix/settings/account/AccountSettingsFragment.kt

```diff
@@ -218,7 +217,57 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
         )
     }
 
-    private fun showPinDialogWarning(newValue: Boolean) {
+    /**
+     * Prompts the user if they do not have a password/pin set up to secure their device, and
+     * updates the state of the sync engine with the new checkbox value.
+     *
+     * Currently used for logins and credit cards.
+     *
+     * @param engine the sync engine whose preference has changed.
```

Comment:
```suggestion
     * @param syncEngine the sync engine whose preference has changed.
```

Maybe we should rename s/engine/syncEngine in `updateSyncEngineState` as well